pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -508,6 +510,28 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 
 		cdLog = cdLog.WithField("job", job.Name)
 
+		// pvc will be nil if log gathering is disabled:
+		if pvc != nil {
+			if err = controllerutil.SetControllerReference(cd, pvc, r.scheme); err != nil {
+				cdLog.WithError(err).Error("error setting controller reference on pvc")
+				return reconcile.Result{}, err
+			}
+
+			existingPVC := &corev1.PersistentVolumeClaim{}
+			err = r.Get(context.TODO(), types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}, existingPVC)
+			if apierrors.IsNotFound(err) {
+				cdLog.WithField("pvc", pvc.Name).Info("creating persistent volume claim")
+				err = r.Create(context.TODO(), pvc)
+				if err != nil {
+					cdLog.WithError(err).Error("error creating pvc")
+					return reconcile.Result{}, err
+				}
+			} else if err != nil {
+				cdLog.WithError(err).Error("error getting persistent volume claim")
+				return reconcile.Result{}, err
+			}
+		}
+
```

Comment:
Can we move this to a function? At this point clusterdeployment_controller.go reconcile function has more code than what should be in a single function. May be moving it to a function would enable us to add tests for it.

Reply:
Will do.

Reply:
> > One question about day-to-day using this. How will we be able to view the PV contents after the logs are collected/saved?
> 
> @dgoodwin I also have the same question.

I've been just rsh'ing into the current install pod. If however a later install completed I think we'll want to do up a script like: https://blog.openshift.com/transferring-files-in-and-out-of-containers-in-openshift-part-3/

Reply:
That article is for copying files *into* the PV, not for copying files out of it. To copy files out, the easiest thing is to do `oc cp ${pod-name}:${path-to-file}`. The pod used could be the current install pod. Or it may be helpful to create a script that starts a pod specifically for mounting the PV to get the desired logs.

Reply:
Yeah this is the approach I wanted from that article, spinning up a dummy pod so we have something we can reliably use to cp or rsh, as the install may have completed successfully after a few failures.